aiohttp/web_response.py

```diff
@@ -420,7 +420,7 @@ async def _prepare_headers(self) -> None:
             # https://datatracker.ietf.org/doc/html/rfc9112#section-6.1-13
             if hdrs.TRANSFER_ENCODING in headers:
                 del headers[hdrs.TRANSFER_ENCODING]
-        elif self.content_length != 0:
+        elif writer.length != 0:
```

Comment:
Would be safer to guard this to fallback to content length if length check is off.  Right now it can't happen though.  Maybe not worth the complexity since it already has a test to prevent regressions